akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -352,4 +357,8 @@ private[http] object OutgoingConnectionBlueprint {
       override def preStart(): Unit = getNextMethod()
     }
   }
+
+  class UnexpectedConnectionClosureException(outstandingResponses: Int)
+    extends RuntimeException(s"The http server closed the connection unexpectedly before delivering responses for $outstandingResponses outstanding requests")
+    with NoStackTrace
```

Comment:
I just kept it from what the default One2OneBidiFlow exceptions do. The stack trace will always be the same. But, I agree, it shouldn't happen to often to make it expensive, and having a stack trace will also help people look into the code (even if it's always the same stack trace). I'll change that.